README.md

```diff
@@ -41,3 +43,45 @@ We use [bunyan](https://github.com/trentm/node-bunyan) for logging:
 * Bunyan by default logs JSON. If you want the json to be pretty printed
   pipe anything that logs into `bunyan` e.g. `LOG_LEVEL=debug grunt test
   | node_modules/bunyan/bin/bunyan`
+
+
+## Architecture
+
+In a nutshell the way the validator works is to take an add-on
+package, extract the metadata from the xpi (zip) format and then
+process the files it finds through various content scanners.
+
+### Scanners
+
+Generally each file-type has a scanner. For example we have a CSS and a
+JavaScript scanner amongst others. Each scanner looks at relevant
+files and passes each file through a parser which then hands off to
+various rules that look for specific things.
+
+### Rules
+
+Rules are generally a single function in a file organized by directories
```

Comment:
No need for "generally" here, as our aim is to always have a one-to-one rule-to-file-to-function ratio. Private functions don't count, so what we're getting at is more:

> Rules are exported via a single function in a single file. A rule can have private functions it uses internally, but rule code should not depend on another rule file and only one function should be exported per file.

That seems oddly worded too, but it's what we want to get at I think?